fabric-tool-attributes-v1/src/main/java/net/fabricmc/fabric/impl/tools/ToolManager.java

```diff
@@ -87,7 +88,9 @@ public static void registerBreakByTool(Block block, Tag<Item> tag, int miningLev
 	}
 
 	private static int getMiningLevel(ItemStack stack) {
-		if (stack.getItem() instanceof ToolItem) {
+		if (stack.getItem() instanceof ToolAttributeHolder) {
+			return ((ToolAttributeHolder) stack.getItem()).getMiningLevel(stack);
+		} else if (stack.getItem() instanceof ToolItem) {
```

Comment:
ToolItem now implements ToolAttributeHolder, so this special case can be removed.

Reply:
…actually, it doesn't. Only MiningToolItem does. I should fix that.

Reply:
You forgot to remove this case